concent_api/core/models.py

```diff
@@ -28,6 +34,25 @@
 from .constants             import MESSAGE_TASK_ID_MAX_LENGTH
 
 
+class SubtaskWithTimingColumnsManager(Manager):
+    def get_queryset(self):
+        bytes_per_sec = settings.MINIMUM_UPLOAD_RATE << 10
```

Comment:
@cameel Do we need comments to this function? If we do, can you write some?

Reply:
Well, indeed, without alignment this looks like a mess. I really hate that our current coding style does not let us make it readable. The original code was quite readable: https://github.com/golemfactory/concent/issues/329#issuecomment-391834635

But this still has a lot of readability problems:
- First, I'd make an exception here and align this part and just add a comment explaining that without alignment this code is unreadable.
- This code is based on `subtask_verification_time()` and `maximum_download_time()` (I even kept the same variable names in my example to make it apparent). A comment saying that is definitely needed. 
    - BTW: you have created tests checking that the results of this query match those functions, right?
- Long mathematical expressions might be more readable broken into multiple lines.
- `time_for_download_file` sounds pretty awkward. This should be called something like `download_deadline`.
    - Actually this variable is not necessary. You know that adding it to `annotate()` creates a new column, right? We don't want any extra columns besides the two that were in the original example.
- Your nesting is off. It's really hard to tell what is nested in what. Is the second `annotate()` nested in the first? I can't tell without counting parentheses. It should be wrapped like this:
    ``` python
    return (super().get_queryset().
        annotate(
            maximum_download_time=Value(settings.DOWNLOAD_LEADIN_TIME) + download_time
        ).
        annotate(
            subtask_verification_time=ExpressionWrapper(subtask_verification_time, output_field=IntegerField())
        ).
        annotate(
            time_for_download_file=ExpressionWrapper(time_for_download_file, output_field=IntegerField())
        )
    )
    ```
    But really, `annotate()` can take multiple parameters. Why don't you do it like in my example?
    ``` python
    return super().get_queryset(). annotate(
        maximum_download_time=Value(settings.DOWNLOAD_LEADIN_TIME) + download_time,
        subtask_verification_time=ExpressionWrapper(subtask_verification_time, output_field=IntegerField()),
        time_for_download_file=ExpressionWrapper(time_for_download_file, output_field=IntegerField()),
    )
    ```
- It may be a good idea to put it in a separate function - that would give it a name. Or maybe even two functions.
    - You could have `add_subtask_verification_time_column()` and `add_maximum_download_time_column()`. The second one would assume that the first one has already been called so that the `subtask_verification_time` column already exists. Then `get_queryset()` would return simply `super().get_queryset().add_subtask_verification_time_column().add_maximum_download_time_column()`
- Finally, you should add a docstring. This docstring should explain what the function does and could also give a  formula for the calculation you're doing.

Reply:
What about splitting it into two functions as suggested in my comment. I think it would go a long way towards making it more readable.